src/Lecture4.hs

```diff
@@ -254,7 +340,10 @@ the file doesn't have any products.
 -}
 
 calculateStats :: String -> String
-calculateStats = error "TODO"
+calculateStats = maybe "file doesn't have any products" (displayStats . combineRows)
+                    . nonEmpty
+                    . mapMaybe parseRow
+                    . lines
```

Comment:
And here I would rather do something like:
```haskell
calculateStats :: String -> String
calculateStats = maybe "file doesn't have any products" (displayStats . combineStats)
                    . nonEmpty
                    . mapMaybe (fmap rowToStats . parseRow)
                    . lines
```
Replacing `combineRows :: NonEmpty Row -> Stats` with `combineStats :: NonEmpty Stats -> Stats`
Seems more natural.